cluster/config.go

```
@@ -83,12 +84,24 @@ func BuildContainerConfig(c dockerclient.ContainerConfig) *ContainerConfig {
 		json.Unmarshal([]byte(labels), &constraints)
 	}
 
+	// parse reschedule policy from labels (ex. docker run --label 'com.docker.swarm.reschedule-policy=on-node-failure')
+	if label, ok := c.Labels[SwarmLabelNamespace+".reschedule-policy"]; ok {
+		reschedulePolicy = label
+	}
+
 	// parse affinities/constraints from env (ex. docker run -e affinity:container==redis -e affinity:image==nginx -e constraint:region==us-east -e constraint:storage==ssd)
 	for _, e := range c.Env {
 		if ok, key, value := parseEnv(e); ok && key == "affinity" {
 			affinities = append(affinities, value)
 		} else if ok && key == "constraint" {
 			constraints = append(constraints, value)
+		} else if ok && key == "reschedule" {
+			if reschedulePolicy == "" {
+				reschedulePolicy = value
+			} else if reschedulePolicy != value {
```

I prefer failing explicitly.